langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/FromBytes.java

```diff
@@ -37,12 +40,19 @@
 //)
 public class FromBytes {
 
+    private static final CharsetDecoder charsetDecoder = StandardCharsets.UTF_8.newDecoder();
+
+    private FromBytes() {
+    }
+
     public static Object fromBytes(BArray bytes) {
         try {
-            return StringUtils.fromString(new String(bytes.getBytes(), StandardCharsets.UTF_8));
-        } catch (Exception e) {
+            String str = charsetDecoder.decode(ByteBuffer.wrap(bytes.getBytes())).toString();
+            charsetDecoder.reset();
+            return StringUtils.fromString(str);
+        } catch (CharacterCodingException e) {
             return ErrorCreator.createError(StringUtils.fromString("FailedToDecodeBytes"),
-                                            StringUtils.fromString(e.getMessage()));
+                    StringUtils.fromString("byte array contains invalid value"));
```

Comment:
Shall we add the error reason and details to `BallerinaErrorReasons` and `RuntimeErrors` respectively.